libbeat/publisher/pipeline/monitoring.go

```diff
@@ -85,19 +96,46 @@ func newMetricsObserver(metrics *monitoring.Registry) *metricsObserver {
 	return &metricsObserver{
 		metrics: metrics,
 		vars: metricsObserverVars{
-			clients: monitoring.NewUint(reg, "clients"), // Gauge
+			// (Gauge) clients measures the number of open pipeline clients.
+			clients: monitoring.NewUint(reg, "clients"),
 
-			events:    monitoring.NewUint(reg, "events.total"),
-			filtered:  monitoring.NewUint(reg, "events.filtered"),
-			published: monitoring.NewUint(reg, "events.published"),
-			failed:    monitoring.NewUint(reg, "events.failed"),
-			dropped:   monitoring.NewUint(reg, "events.dropped"),
-			retry:     monitoring.NewUint(reg, "events.retry"),
+			// events.total counts all created events.
+			eventsTotal: monitoring.NewUint(reg, "events.total"),
 
-			queueACKed:     monitoring.NewUint(reg, "queue.acked"),
+			// (Gauge) events.active measures events that have been created, but have
+			// not yet been failed, filtered, or acked/dropped.
+			activeEvents: monitoring.NewUint(reg, "events.active"),
+
+			// events.filtered counts events that were filtered by processors before
+			// being sent to the queue.
+			eventsFiltered: monitoring.NewUint(reg, "events.filtered"),
+
+			// events.failed counts events that were rejected by the queue, or that
+			// were sent via an already-closed pipeline client.
+			eventsFailed: monitoring.NewUint(reg, "events.failed"),
+
+			// events.published counts events that were accepted by the queue.
+			eventsPublished: monitoring.NewUint(reg, "events.published"),
+
+			// events.retry counts events that an output worker sent back to be
+			// retried.
+			eventsRetry: monitoring.NewUint(reg, "events.retry"),
+
+			// events.dropped counts events that were dropped because errors from
+			// the output workers exceeded the configured maximum retry count.
+			eventsDropped: monitoring.NewUint(reg, "events.dropped"),
+
+			// (Gauge) queue.max_events measures the maximum number of events the
+			// queue will accept, or 0 if there is none.
 			queueMaxEvents: monitoring.NewUint(reg, "queue.max_events"),
 
-			activeEvents:     monitoring.NewUint(reg, "events.active"), // Gauge
+			// queue.acked counts events that have been acknowledged by the output
+			// workers. This includes events that were dropped for fatal errors,
+			// which are also reported in events.dropped.
+			queueACKed: monitoring.NewUint(reg, "queue.acked"),
+
+			// (Gauge) queue.filled.pct.events measures the fraction (from 0 to 1)
```

Comment:
Can we drop the `.events` from this, ahead of adding support for the queue size in bytes?

People will want to alert on this, they shouldn't have to define it twice.

Reply:
Already part of https://github.com/elastic/beats/pull/39774